silence/silence_test.go

```diff
@@ -109,6 +109,119 @@ func TestSilencesGC(t *testing.T) {
 	require.Equal(t, want, s.st)
 }
 
+func TestSilenceGCOverTime(t *testing.T) {
```

Comment:
It seems to me that we should fold this test into `TestSilenceGC`?

Reply:
That seems reasonable to me as well - I left them both in just to avoid adding additional mental overhead when looking at the new test. The first test case should be exactly the same as `TestSilenceGC`